src/test/java/org/jenkins/plugins/lockableresources/TryLockStepTest.java

```diff
@@ -0,0 +1,75 @@
+package org.jenkins.plugins.lockableresources;
+
+import hudson.model.*;
+import java.util.Arrays;
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+
+public class TryLockStepTest extends LockStepTestBase {
+
+  private static final String SCRIPT_TRY_DONT_RELEASE =
+      "if (tryLock(resource: 'resource1')) {\n"
+          + "  echo 'Returned True'\n"
+          + "} else {\n"
+          + "  echo 'Returned False'\n"
+          + "}";
+  private static final String SCRIPT_TRY_AND_RELEASE =
+      "if (tryLock(resource: 'resource1')) {\n"
+          + "  echo 'Returned True'\n"
+          + "  releaseLock(resource: 'resource1')\n"
+          + "} else {"
+          + "  echo 'Returned False'\n"
+          + "}";
+  @Rule public JenkinsRule j = new JenkinsRule();
+
+  @Test
+  public void reserveAvailableResource() throws Exception {
+    LockableResourcesManager lm = LockableResourcesManager.get();
+    lm.createResourceWithLabel("resource1", "label1");
+
+    WorkflowJob p = j.jenkins.createProject(WorkflowJob.class, "p");
+    p.setDefinition(new CpsFlowDefinition(SCRIPT_TRY_DONT_RELEASE));
+
+    WorkflowRun b1 = p.scheduleBuild2(0).waitForStart();
+
+    j.assertBuildStatusSuccess(j.waitForCompletion(b1));
+
+    j.assertLogContains("Returned True", b1);
```

Comment:
You need to use a `RunListener` or similar to ensue that any locks acquired by a build are released when it exits even if the user forgot to use the release step (or forgot to put it in a `finally` block, or the CPS VM crashed and the build did not exit via the Groovy call stack).

Reply:
I made some manual tests and as far as I can see, the locks are released by LockRunListener:
https://github.com/jenkinsci/lockable-resources-plugin/blob/master/src/main/java/org/jenkins/plugins/lockableresources/queue/LockRunListener.java

Reply:
OK. Worth adding a test for it I think.